code/modules/mob/living/simple_animal/hostile/hivebot.dm

```diff
@@ -23,18 +23,47 @@
 	faction = list("hivebot")
 	check_friendly_fire = 1
 	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	possible_a_intents = list(INTENT_HELP, INTENT_GRAB, INTENT_DISARM, INTENT_HARM)
 	minbodytemp = 0
-	speak_emote = list("states")
+	verb_say = "states"
+	verb_ask = "queries"
+	verb_exclaim = "declares"
+	verb_yell = "alarms"
+	bubble_icon = "machine"
+	speech_span = SPAN_ROBOT
 	gold_core_spawnable = HOSTILE_SPAWN
 	del_on_death = 1
 	loot = list(/obj/effect/decal/cleanable/robot_debris)
+	var/alert_light
 
 	do_footstep = TRUE
 
 /mob/living/simple_animal/hostile/hivebot/Initialize()
 	. = ..()
 	deathmessage = "[src] blows apart!"
 
+/mob/living/simple_animal/hostile/hivebot/Aggro()
+	. = ..()
+	a_intent_change(INTENT_HARM)
+	if(prob(5))
+		say(pick("INTRUDER DETECTED!", "CODE 7-34.", "101010!!"), forced = type)
+
+/mob/living/simple_animal/hostile/hivebot/LoseAggro()
+	. = ..()
+	a_intent_change(INTENT_HELP)
+
+/mob/living/simple_animal/hostile/hivebot/a_intent_change(input as text)
+	. = ..()
+	update_icons()
+
+/mob/living/simple_animal/hostile/hivebot/update_icons()
+	qdel(alert_light)

```

Comment:
QDEL_NULL, it won't get cleanly deleted if you change to help intent.

Reply:
Did that. May I ask why it's needed?